### PR TITLE
Closets/Bags auto close on closet closing now - Fixes #8821

### LINKED ISSUE #8821: You can have a bag's inventory open while it goes into a bluespace body bag
What this means is you can then put the bluespace body bag inside of the bag, which is inside of the bluespace body bag

This makes the bbg and the contents within unaccessible

UNLESS

When deleting the person that did this, the body bag's contents to spill out where they were deleted

### PR DESCRIPTION
**Pull request by JamieD1**
Fixes #8821 

Closet/Bags auto close on closet closing now 

#### Changelog

:cl:  
bugfix: 
/:cl:

### CODE REVIEW

**Comment by JamieD1** [comment]
Tog ETA on merge: 18th
Jamie ETA on merge: 11th
Should be merged: 8th

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -209,6 +210,11 @@
 	else
 		return open(user)
 
+/obj/structure/closet/proc/close_storage(mob/living/user)
+	for(var/obj/item/storage/opened in contents)
+		var/datum/component/storage/closeall = opened.GetComponent(/datum/component/storage)
+		closeall.close_all()
```
Comment (alexkar598) [inline code/game/objects/structures/crates_lockers/closets.dm]:
```suggestion
	for(var/obj/object in contents)
		var/datum/component/storage/closeall = opened.GetComponent(/datum/component/storage)
		if(closeall)
			closeall.close_all()
```

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/crates_lockers/closets.dm (hunk 1 above)
Comment (alexkar598) [reply code/game/objects/structures/crates_lockers/closets.dm]: Storage is a component... the point is to not need to rely on typepath for behaviour

**Review by alexkar598 — APPROVED** [review]

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 2]
```diff
@@ -209,6 +210,12 @@
 	else
 		return open(user)
 
+/obj/structure/closet/proc/close_storage(mob/living/user)
+	for(var/obj/object in contents)
+		var/datum/component/storage/closeall = opened.GetComponent(/datum/component/storage)
```
Comment (alexkar598) [inline code/game/objects/structures/crates_lockers/closets.dm]:
```suggestion
		var/datum/component/storage/closeall = object.GetComponent(/datum/component/storage)
```

imsmartok

**Review by alexkar598 — CHANGES_REQUESTED** [review]

**Pull request merged**